### ISSUE TITLE
OEC2018

### ISSUE THREAD

**Issue by NikhilPeri** [NikhilPeri/resume#1]
A Hacky chat solution for [oec 2018](https://oec-2018.herokuapp.com)

**Comment by MooseV2** [comment]
First post

> You can also edit other comments on github issues
> Dont do this, play nice
> - Nik

**Comment by NikhilPeri** [comment]
UPDATE 2018-01-26 20:58:26
My internet is shit trading will be delayed by 15 min

**Comment by ColinNowers** [comment]
Still can't buy any updates?

**Comment by dfsisinni** [comment]
we are unable to buy stocks rn :/

**Comment by WxnChen11** [comment]
historical price doesn't seem to be working either, only showing one price in list

**Comment by NikhilPeri** [comment]
WE ARE LIVE! any issues now should be real

**Comment by ethernet8023** [comment]
buying stocks still throws a 500 for me

**Comment by dfsisinni** [comment]
<!DOCTYPE html><html><head>  <title>We're sorry, but something went wrong (500)

**Comment by dfsisinni** [comment]
^when buying stocks

**Comment by dfsisinni** [comment]
![image](https://user-images.githubusercontent.com/13513321/35467646-37d6b76e-02df-11e8-9076-3831a01e5460.png)

**Comment by NikhilPeri** [comment]
^ looking into this, my internet is spotty so I will see if I can recreate it

**Comment by dfsisinni** [comment]
^happens every time we run a buy transaction

**Comment by AarushiSab** [comment]
We are experiencing the same problem

**Comment by z3t0** [comment]
I keep getting an internal 500 error when we try to purchase

**Comment by arilotter** [comment]
JavaScript in the browser can't call the API due to CORS. You can fix this by sending an `Access-Control-Allow-Origin` header with value `*`. The exact error is: `index.html:1 Failed to load https://oec-2018.herokuapp.com/api/stock/list: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'null' is therefore not allowed access. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

**Comment by CSchank** [comment]
^^ The CORS error is because the server is not sending back a proper request to let the browser know that the request was legit.

**Comment by arilotter** [comment]
@CSchank Yeah, I'm asking Nikhil to add access-control-allow-origin headers so browser javascript can access the API without a backend

**Comment by kimwoo11** [comment]
When selling stock, is it GET /api/sell?ticker=[stock_identifier]&shares=[num_shares]key=[api_key] or 
GET /api/sell?ticker=[stock_identifier]&shares=[num_shares]**&**key=[api_key]

**Comment by NikhilPeri** [comment]
Not a CORS problem, it was server side...fix just went out and 🎩 'd it
we should be rolling

**Comment by ethernet8023** [comment]
@NikhilPeri Unrelated to selling, I'd like if you added CORS headers so I can call the API from the browser.

**Comment by arilotter** [comment]
The 'cash' value in the `account` endpoint does not update upon buying stocks.

**Comment by ethernet8023** [comment]
The API is bugged: it does not show the actual value of remaining cash when you call `/api/account`, instead it always shows 10000000.

**Comment by ethernet8023** [comment]
Our account was reset, and all our holdings deleted. ????

**Comment by zako160** [comment]
^ same

**Comment by CSchank** [comment]
Can we get an api for only current amount and historical as a separate thing?

**Comment by dfsisinni** [comment]
our holdings keep disappearing

**Comment by NikhilPeri** [comment]
THERE WAS A MAJOR BUG IN STORING TRANSACTIONS...mostly cause by me not storing them, lol I have reset all account holdings and cash. and fixed the bug

**Comment by ColinNowers** [comment]
@NikhilPeri are you ranking the performance on cash value right now?

**Comment by NikhilPeri** [comment]
No it is ranked by `total_value = holding + cash value`

**Comment by z3t0** [comment]
There is a non conformance to the docs for one of the API call results. 'historical_prices' is actually returned as 'historical_price'.

**Comment by wsikk** [comment]
whats going on with the standings lol

**Comment by z3t0** [comment]
is the historical price at index 0 the most recent or oldest

**Comment by AcharyaS97** [comment]
Im getting a 404 error trying to buy

**Comment by NikhilPeri** [comment]
deploying the following fixes: 
 - working on a fix for the standings page i think my x and y axis are swapped 🤷‍♂️ 
 - and a doc fix for @z3t0 ❤️ 

The historical price is index 0 is oldest

**Comment by SamTheOnionMan** [comment]
It's down right now too, can't access the main page to see the graph or anything.

**Comment by NikhilPeri** [comment]
can everyone add time outs between price calls...I have 2 webservers and I dont want to add a third

**Comment by Mogball** [comment]
90 stocks * 14 teams, checking prices every minute. Please Add a third webserver

edit: hotel wifi is making this hard enough already -- just bought another 1gb of data ecksdee

**Comment by NikhilPeri** [comment]
you cost me so much money... added a 3rd the down time was because of this

also to be fair i am currently using the shittiest servers heroku offers

**Comment by SamTheOnionMan** [comment]
What were the deleted comments about 🤔

**Comment by Mogball** [comment]
Get uOttawa to reimburse ;^)

**Comment by DanielCefaratti** [comment]
Do historical prices always include the current tick?

**Comment by NikhilPeri** [comment]
unless there is a weird race condition it should

**Comment by NikhilPeri** [comment]
the home view was kinda fucked...I think i fixed it 🤷‍♂️

**Comment by SamTheOnionMan** [comment]
not investing everything I had into JOE - worst mistake of my life

**Comment by CSchank** [comment]
It would be nice if this hotel had good internet....

**Comment by zako160** [comment]
is it me or is the market not updating

**Comment by ethernet8023** [comment]
it just updated, but it's definitely slow

**Comment by NikhilPeri** [comment]
Sorry I had to pause the simulation to scale up the sidekiq workers....but my wifi craped out so i couldnt restart it

done now! and added some more stocks TOTAL COUNT IS 160

**Comment by DanielCefaratti** [comment]
You wound me

**Comment by DanielCefaratti** [comment]
Do all stocks always fluctuate every tick? Or can they remain static?

**Comment by SamTheOnionMan** [comment]
they haven't changed in like 10 minutes

**Comment by SamTheOnionMan** [comment]
at least my portfolio hasnt

**Comment by SamTheOnionMan** [comment]
I need to talk to my accountant about that

**Comment by Mogball** [comment]
It takes me 5 minutes to query the price of all stocks, can you add an endpoint that allows us to simultaneously retrieve all stock prices? This reduces the load on both our ends

**Comment by DanielCefaratti** [comment]
Well I don’t know if that’s on purpose or because of the internet problems, I’d like confirmation

**Comment by Mogball** [comment]
I'm ticking on multiple threads and every 5-6 seconds it'll hang for 15 seconds then go back at it again

**Comment by NikhilPeri** [comment]
Maybe you should not query the whole dataset every-time and make some smart decisions

**Comment by Mogball** [comment]
This is "big data" problem, more data = more money

**Comment by SamTheOnionMan** [comment]
it takes like 10 seconds to query with this internet and we cant just watch like 3 stocks

**Comment by NikhilPeri** [comment]
right now I am runnning 28 requests per second and response time is about 2 sec per request
I AM WORKING ON A PR TO THROTTLE PEOPLE MAKING LOTS OF REQUESTS this falls under the ddos rule so I suggest you re-think what data you "need" or prepare to see a slow down

**Comment by ethernet8023** [comment]
Stocks have different lengths of historical price arrays - for example, YIR has 135 elements and WHT has 142.

**Comment by NikhilPeri** [comment]
also no one said yo have to run this on you laptop....there were points for designing a system which can actually be deployed

☁️

**Comment by Mogball** [comment]
I brought an 8-core (x2 for hyperthreading) desktop

**Comment by NikhilPeri** [comment]
@arilotter stocks can be added an removed from the exchange depending on price, you code should be reseliant to this

**Comment by michaeluoft12** [comment]
The challenge should be to redesign the API

**Comment by ethernet8023** [comment]
@NikhilPeri is there somewhere I can ask you a question privately?

**Comment by NikhilPeri** [comment]
@michaeluoft12 you team can barely make 10k let alone a trading platform
shout out to @Guelf you guys are making lots of trades and lots of $$

**Comment by NikhilPeri** [comment]
@arilotter I am doing rounds now...but email me nperi104@uottawa.ca

**Comment by AarushiSab** [comment]
We buy stalks it says true, but our account doesn't change

**Comment by NikhilPeri** [comment]
also fixed the dashboard http://oec-2018.herokuapp.com/ the data before 9 is not really a part of the sim

@AarushiSab what team are you?

**Comment by mshoe** [comment]
I'd just like to point out @michaeluoft12 is not Michael Xu from uoft. That is me. Also not someone from my team either haha..

**Comment by AarushiSab** [comment]
YorkU

**Comment by AarushiSab** [comment]
We fixed it

**Comment by dfsisinni** [comment]
we are unable to trade right now - 500s for sell orders @NikhilPeri

**Comment by DanielCefaratti** [comment]
We just got a buy order through

**Comment by akovo** [comment]
![image](https://user-images.githubusercontent.com/28958022/35469096-ae84ae68-02fb-11e8-8b94-ada262737952.png)

What happens when we try to put a sell order thru when we own more than the selling quantity
@NikhilPeri

**Comment by dfsisinni** [comment]
can't make trades still @NikhilPeri

**Comment by ethernet8023** [comment]
@dfsisinni I think it's the hotel wifi, try using your phone as a hotspot

**Comment by akovo** [comment]
We were using a hotspot @NikhilPeri  Anything you can see on your end

**Comment by wsikk** [comment]
Will the judges be given our code before the presentation? How/when are they marking our code quality @NikhilPeri

**Comment by NikhilPeri** [comment]
@kimwoo11 yes all code must be submited via git repository or email at nperi104@uottawa.ca and it will be marked by reading the code and trying to understand what it does....do class names make sense to functions do what they say they do...no need to write docs, but maybe a small readme to explain your algo would be nice

**Comment by NikhilPeri** [comment]
as for the people who cannot trade are these still going through expect 15s delay

<img width="486" alt="screen shot 2018-01-27 at 1 40 25 am" src="https://user-images.githubusercontent.com/11625693/35469461-213ba8f6-0303-11e8-8980-45d58a015af1.png">

**Comment by NikhilPeri** [comment]
@akovo double check you stock is still on the exchange...things a falling an being deleted and some new tickers are being generated

**Comment by dfsisinni** [comment]
stlll can't sell @NikhilPeri

**Comment by akovo** [comment]
@NikhilPeri  our holdings still have market value and we cant sell to buy because we dont have cash, we keep getting 500 errors. Even with 15s these things still exist

**Comment by NikhilPeri** [comment]
tell me what rooms your in

**Comment by NikhilPeri** [comment]
@akovo @dfsisinni ^

**Comment by akovo** [comment]
829

**Comment by NikhilPeri** [comment]
REMINDER PLEASE EMAIL OR SEND A GITHUB LINK 
ALSO INCLUDE YOU SLIDE DECK!!!!

**Comment by ethernet8023** [comment]
uhhhhh it's 3am and my stocks are still appreciating should the sim be off now?

**Comment by NikhilPeri** [comment]
yup we are off! I tried to just make everything crash for the last 5 min ....but i acidently bade everything boom lol

**Comment by ethernet8023** [comment]
@NikhilPeri will the server be up again tomorrow for presentations?

**Comment by NikhilPeri** [comment]
I can try ... I will open up signups on a different exchange if I have time tonight
@arilotter

**Comment by LemonPi** [comment]
@NikhilPeri One of the judges mentioned trying out linear regression. I implemented a pretty simple one and it consistently (with different parameters) scores $100mil+ when starting to trade on turn 200. Momentum (from autocorrelation) really dominated...

Check it out https://github.com/LemonPi/oec-2018 with `python simulator.py`
